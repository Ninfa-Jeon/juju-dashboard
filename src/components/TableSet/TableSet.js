import React from "react";

const tableSet = {
  blocked: [
    {
      model: "1cdk_default",
      owner: "IS-team",
      configuration: 21,
      cloud: "AWS",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    },
    {
      model: "1my_openstack",
      owner: "IS-team",
      configuration: 19,
      cloud: "Google",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    },
    {
      model: "1kubernetes_core",
      owner: "Saad",
      configuration: 16,
      cloud: "AWS",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    },
    {
      model: "1hapood",
      owner: "Asad",
      configuration: 25,
      cloud: "Google",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    }
  ],
  deploying: [
    {
      model: "2hapood",
      owner: "Wasif",
      configuration: 21,
      cloud: "AWS",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    },
    {
      model: "3hapood",
      owner: "Ali",
      configuration: 19,
      cloud: "Google",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    },
    {
      model: "4hapood",
      owner: "Saad",
      configuration: 16,
      cloud: "AWS",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    },
    {
      model: "5hapood",
      owner: "Asad",
      configuration: 25,
      cloud: "Google",
      region: "eu-west-1",
      credential: "clean-algebra-1",
      controller: "prodstack-is",
      last_updated: "2019.01.01"
    }
  ]
};

const returnTableHeader = table => {
  const header = Object.keys(table[0]);
  return header.map(key => {
    return <th key={key.toString()}>{key.toUpperCase()}</th>;
  });
};

const returnTableBodyData = row => {
  const td = Object.values(row);
  return td.map(value => {
    return <td key={value.toString()}>{value}</td>;
  });
};

const returnTableBodyRows = table => {
  return table.map(row => {
    return <tr key={row.model}>{returnTableBodyData(row)}</tr>;
  });
};

const renderTables = tablesObj => {
  const tables = Object.keys(tablesObj);
  return tables.map((key, i) => (
    <table key={tables[i]}>
      <thead>
        <tr>{returnTableHeader(tablesObj[key])}</tr>
      </thead>
      <tbody>{returnTableBodyRows(tablesObj[key])}</tbody>
    </table>
  ));
};

export default function TableSet() {
  return renderTables(tableSet);
}
