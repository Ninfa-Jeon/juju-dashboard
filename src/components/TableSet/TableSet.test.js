import React from "react";
import { mount } from "enzyme";

import TableSet from "./TableSet";

describe("tableSet", () => {
  it("renders without crashing and matches snapshot", () => {
    const wrapper = mount(<TableSet />);
    expect(wrapper).toMatchInlineSnapshot();
  });
});
