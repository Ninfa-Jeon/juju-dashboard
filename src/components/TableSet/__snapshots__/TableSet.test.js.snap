// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Set renders without crashing and matches snapshot 1`] = `
<TableSet>
  <Table
    key="blocked"
  >
    <table
      className=""
      role="grid"
    >
      <thead>
        <TableRow>
          <tr
            role="row"
          >
            <TableHeader
              key="model"
            >
              <th
                role="columnheader"
              >
                MODEL
              </th>
            </TableHeader>
            <TableHeader
              key="owner"
            >
              <th
                role="columnheader"
              >
                OWNER
              </th>
            </TableHeader>
            <TableHeader
              key="configuration"
            >
              <th
                role="columnheader"
              >
                CONFIGURATION
              </th>
            </TableHeader>
            <TableHeader
              key="cloud"
            >
              <th
                role="columnheader"
              >
                CLOUD
              </th>
            </TableHeader>
            <TableHeader
              key="region"
            >
              <th
                role="columnheader"
              >
                REGION
              </th>
            </TableHeader>
            <TableHeader
              key="credential"
            >
              <th
                role="columnheader"
              >
                CREDENTIAL
              </th>
            </TableHeader>
            <TableHeader
              key="controller"
            >
              <th
                role="columnheader"
              >
                CONTROLLER
              </th>
            </TableHeader>
            <TableHeader
              key="last_updated"
            >
              <th
                role="columnheader"
              >
                LAST_UPDATED
              </th>
            </TableHeader>
          </tr>
        </TableRow>
      </thead>
      <tbody>
        <TableRow
          key="1cdk_default"
        >
          <tr
            role="row"
          >
            <TableCell
              key="1cdk_default"
            >
              <td
                className=""
                role="gridcell"
              >
                1cdk_default
              </td>
            </TableCell>
            <TableCell
              key="IS-team"
            >
              <td
                className=""
                role="gridcell"
              >
                IS-team
              </td>
            </TableCell>
            <TableCell
              key="21"
            >
              <td
                className=""
                role="gridcell"
              >
                21
              </td>
            </TableCell>
            <TableCell
              key="AWS"
            >
              <td
                className=""
                role="gridcell"
              >
                AWS
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
        <TableRow
          key="1my_openstack"
        >
          <tr
            role="row"
          >
            <TableCell
              key="1my_openstack"
            >
              <td
                className=""
                role="gridcell"
              >
                1my_openstack
              </td>
            </TableCell>
            <TableCell
              key="IS-team"
            >
              <td
                className=""
                role="gridcell"
              >
                IS-team
              </td>
            </TableCell>
            <TableCell
              key="19"
            >
              <td
                className=""
                role="gridcell"
              >
                19
              </td>
            </TableCell>
            <TableCell
              key="Google"
            >
              <td
                className=""
                role="gridcell"
              >
                Google
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
        <TableRow
          key="1kubernetes_core"
        >
          <tr
            role="row"
          >
            <TableCell
              key="1kubernetes_core"
            >
              <td
                className=""
                role="gridcell"
              >
                1kubernetes_core
              </td>
            </TableCell>
            <TableCell
              key="Saad"
            >
              <td
                className=""
                role="gridcell"
              >
                Saad
              </td>
            </TableCell>
            <TableCell
              key="16"
            >
              <td
                className=""
                role="gridcell"
              >
                16
              </td>
            </TableCell>
            <TableCell
              key="AWS"
            >
              <td
                className=""
                role="gridcell"
              >
                AWS
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
        <TableRow
          key="1hapood"
        >
          <tr
            role="row"
          >
            <TableCell
              key="1hapood"
            >
              <td
                className=""
                role="gridcell"
              >
                1hapood
              </td>
            </TableCell>
            <TableCell
              key="Asad"
            >
              <td
                className=""
                role="gridcell"
              >
                Asad
              </td>
            </TableCell>
            <TableCell
              key="25"
            >
              <td
                className=""
                role="gridcell"
              >
                25
              </td>
            </TableCell>
            <TableCell
              key="Google"
            >
              <td
                className=""
                role="gridcell"
              >
                Google
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
      </tbody>
    </table>
  </Table>
  <Table
    key="deploying"
  >
    <table
      className=""
      role="grid"
    >
      <thead>
        <TableRow>
          <tr
            role="row"
          >
            <TableHeader
              key="model"
            >
              <th
                role="columnheader"
              >
                MODEL
              </th>
            </TableHeader>
            <TableHeader
              key="owner"
            >
              <th
                role="columnheader"
              >
                OWNER
              </th>
            </TableHeader>
            <TableHeader
              key="configuration"
            >
              <th
                role="columnheader"
              >
                CONFIGURATION
              </th>
            </TableHeader>
            <TableHeader
              key="cloud"
            >
              <th
                role="columnheader"
              >
                CLOUD
              </th>
            </TableHeader>
            <TableHeader
              key="region"
            >
              <th
                role="columnheader"
              >
                REGION
              </th>
            </TableHeader>
            <TableHeader
              key="credential"
            >
              <th
                role="columnheader"
              >
                CREDENTIAL
              </th>
            </TableHeader>
            <TableHeader
              key="controller"
            >
              <th
                role="columnheader"
              >
                CONTROLLER
              </th>
            </TableHeader>
            <TableHeader
              key="last_updated"
            >
              <th
                role="columnheader"
              >
                LAST_UPDATED
              </th>
            </TableHeader>
          </tr>
        </TableRow>
      </thead>
      <tbody>
        <TableRow
          key="2hapood"
        >
          <tr
            role="row"
          >
            <TableCell
              key="2hapood"
            >
              <td
                className=""
                role="gridcell"
              >
                2hapood
              </td>
            </TableCell>
            <TableCell
              key="Wasif"
            >
              <td
                className=""
                role="gridcell"
              >
                Wasif
              </td>
            </TableCell>
            <TableCell
              key="21"
            >
              <td
                className=""
                role="gridcell"
              >
                21
              </td>
            </TableCell>
            <TableCell
              key="AWS"
            >
              <td
                className=""
                role="gridcell"
              >
                AWS
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
        <TableRow
          key="3hapood"
        >
          <tr
            role="row"
          >
            <TableCell
              key="3hapood"
            >
              <td
                className=""
                role="gridcell"
              >
                3hapood
              </td>
            </TableCell>
            <TableCell
              key="Ali"
            >
              <td
                className=""
                role="gridcell"
              >
                Ali
              </td>
            </TableCell>
            <TableCell
              key="19"
            >
              <td
                className=""
                role="gridcell"
              >
                19
              </td>
            </TableCell>
            <TableCell
              key="Google"
            >
              <td
                className=""
                role="gridcell"
              >
                Google
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
        <TableRow
          key="4hapood"
        >
          <tr
            role="row"
          >
            <TableCell
              key="4hapood"
            >
              <td
                className=""
                role="gridcell"
              >
                4hapood
              </td>
            </TableCell>
            <TableCell
              key="Saad"
            >
              <td
                className=""
                role="gridcell"
              >
                Saad
              </td>
            </TableCell>
            <TableCell
              key="16"
            >
              <td
                className=""
                role="gridcell"
              >
                16
              </td>
            </TableCell>
            <TableCell
              key="AWS"
            >
              <td
                className=""
                role="gridcell"
              >
                AWS
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
        <TableRow
          key="5hapood"
        >
          <tr
            role="row"
          >
            <TableCell
              key="5hapood"
            >
              <td
                className=""
                role="gridcell"
              >
                5hapood
              </td>
            </TableCell>
            <TableCell
              key="Asad"
            >
              <td
                className=""
                role="gridcell"
              >
                Asad
              </td>
            </TableCell>
            <TableCell
              key="25"
            >
              <td
                className=""
                role="gridcell"
              >
                25
              </td>
            </TableCell>
            <TableCell
              key="Google"
            >
              <td
                className=""
                role="gridcell"
              >
                Google
              </td>
            </TableCell>
            <TableCell
              key="eu-west-1"
            >
              <td
                className=""
                role="gridcell"
              >
                eu-west-1
              </td>
            </TableCell>
            <TableCell
              key="clean-algebra-1"
            >
              <td
                className=""
                role="gridcell"
              >
                clean-algebra-1
              </td>
            </TableCell>
            <TableCell
              key="prodstack-is"
            >
              <td
                className=""
                role="gridcell"
              >
                prodstack-is
              </td>
            </TableCell>
            <TableCell
              key="2019.01.01"
            >
              <td
                className=""
                role="gridcell"
              >
                2019.01.01
              </td>
            </TableCell>
          </tr>
        </TableRow>
      </tbody>
    </table>
  </Table>
</TableSet>
`;
